From f9a5c90ca67303f600d9d1ece71d7cb036504e6a Mon Sep 17 00:00:00 2001
From: Morb0 <14136326+Morb0@users.noreply.github.com>
Date: Sun, 20 Aug 2023 00:32:05 +0300
Subject: [PATCH] fix

---
 SS14.Admin/Startup.cs | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/SS14.Admin/Startup.cs b/SS14.Admin/Startup.cs
index f2a0a19..2de9773 100644
--- a/SS14.Admin/Startup.cs
+++ b/SS14.Admin/Startup.cs
@@ -67,6 +67,18 @@ namespace SS14.Admin
                     options.Scope.Add("openid");
                     options.Scope.Add("profile");
                     options.GetClaimsFromUserInfoEndpoint = true;
+                    options.NonceCookie.SameSite = SameSiteMode.None;
+                    options.NonceCookie.SecurePolicy = CookieSecurePolicy.Always;
+                    options.CorrelationCookie.SameSite = SameSiteMode.None;
+                    options.CorrelationCookie.SecurePolicy = CookieSecurePolicy.Always;
+
+                    options.Events.OnRedirectToIdentityProvider = async ctx =>
+                    {
+                        if (ctx.ProtocolMessage.RedirectUri.StartsWith("http://"))
+                            ctx.ProtocolMessage.RedirectUri =
+                                ctx.ProtocolMessage.RedirectUri.Replace("http://", "https://");
+                        await Task.FromResult(0);
+                    };
 
                     options.Events.OnTokenValidated = async ctx =>
                     {
@@ -111,7 +123,6 @@ namespace SS14.Admin
             }
 
             app.UseAuthentication();
-            app.UseHttpsRedirection();
             app.UseStaticFiles();
 
             app.UseRouting();
-- 
2.40.1.windows.1

